### PR TITLE
feat(gumby): run mvnw if mvn fails

### PR DESCRIPTION
**Pull request by dhasani23**
## Problem

When `mvn` fails, we want to run `mvnw` because that may succeed, since the user could have the Maven wrapper in their workspace despite not having Maven installed.

## Solution

Run `mvnw` when `mvn` fails. Some smaller things done in this PR: 1) slight renaming and 2) collect input in a simpler way (no need to use `MultiStepInputFlowController` since we only collect 1 piece of input).

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/service/transformByQHandler.ts [hunk 1]
```diff
@@ -326,8 +338,7 @@ function getProjectDependencies(modulePath: string): string[] {
     const spawnResult = spawnSync(baseCommand, args, { cwd: modulePath, shell: true, encoding: 'utf-8' })
```
[the remark this answers is not in the document]
Comment (justinmk3) [reply src/codewhisperer/service/transformByQHandler.ts:338]: If you are passing a hardcoded, relative path, `shell: true` has less risk. But it still means that the system-configured shell is involved, which means unusual systems may cause problems and will be hard to troubleshoot.

**Review by justinmk3 — APPROVED** [review]
This is ever the problem with this kind of design: you are re-inventing a build system / IDE. The edge cases will continue to accumulate. If there is a way to re-use vscode's java plugin instead, that could avoid all of this toil.

**Comment by justinmk3** [comment]
waiting for gumby team approval/review

File: src/codewhisperer/service/transformByQHandler.ts [hunk 2]
```diff
@@ -310,12 +310,24 @@ function getFilesRecursively(dir: string, isDependenciesFolder: boolean): string
...
 
-function getProjectDependencies(modulePath: string): string[] {
+function getProjectDependencies(buildCommand: CodeTransformMavenBuildCommand, modulePath: string): string[] {
     // Make temp directory
     const folderName = `${CodeWhispererConstants.dependencyFolderName}${Date.now()}`
     const folderPath = path.join(os.tmpdir(), folderName)
 
-    const baseCommand = 'mvn'
+    let baseCommand = buildCommand as string
+    if (baseCommand === 'mvnw') {
```
Comment (damntrecky) [inline src/codewhisperer/service/transformByQHandler.ts:319]: Question: It looks like IntelliJ we run `./mvnw` first and `mvn` second. I think we should align on the order. See IntelliJ Implementation in [CodeModernizerSessionContext.kt](https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt#L190)

**Review by damntrecky — APPROVED** [review]
Tentative approval. Please review the order in which we run these commands to align with IntelliJ.

File: src/codewhisperer/service/transformByQHandler.ts (hunk 2 above)
Comment (dhasani23) [reply src/codewhisperer/service/transformByQHandler.ts:319]: Agreed & fixed

**Comment by damntrecky** [comment]
> waiting for gumby team approval/review

Approved 👍 🟢

**Pull request merged**